### PR TITLE
Fixed audio messages crash when sent

### PR DESCRIPTION
**Pull request by hamzahass**
This bug was related to the fact that the audio messages were stored on the external storage of the device (SD Card), which was okay in the case of the emulator, but for actual devices, not all of them have external storage.

### CODE REVIEW

File: app/src/main/java/com/github/brugapp/brug/ui/ChatActivity.kt [hunk 1]
```diff
@@ -266,9 +249,8 @@ class ChatActivity : AppCompatActivity() {
             audioRecMessage.visibility = View.GONE
             messageLayout.visibility = View.VISIBLE
             recordButton.visibility = View.VISIBLE
-            model.setListenForRecord(recordButton, false)
 
-            model.sendAudio()
+            model.sendAudio(this, convID, firestore, firebaseAuth, firebaseStorage)
```
Comment (pmcintyr) [inline app/src/main/java/com/github/brugapp/brug/ui/ChatActivity.kt:253]: Hi, I saw your refactoring to merge setListenForRecord and sendAudio into 1 function. Are we sure we never want to use setListenForRecord without sendAudio? If this were the case, separating the methods could be an advantage. If not, no problem, just explain to me why this is the case.

File: app/src/test/java/com/github/brugapp/brug/MessageTest.kt [hunk 2]
```diff
@@ -246,8 +247,9 @@ class MessageTest {
         )
 
         val audioUrl = "/path/to/dummy/audio.mp3"
+        val audioPath = "/path/to/dummy/audio.mp3"
```
Comment (pmcintyr) [inline app/src/test/java/com/github/brugapp/brug/MessageTest.kt:250]: I was wondering if we need to keep both variables here(audioUrl & audioPath), considering they contain the same dummy string. Can't we just keep 1?

File: app/src/test/java/com/github/brugapp/brug/MessageTest.kt [hunk 3]
```diff
@@ -263,9 +265,10 @@ class MessageTest {
         ))
         val body = "TESTMESSAGE"
         val audioUrl = "/path/to/dummy/audio.mp3"
```
Comment (pmcintyr) [inline app/src/test/java/com/github/brugapp/brug/MessageTest.kt:267]: Same thing here for audioUrl & audioPath

File: app/src/main/java/com/github/brugapp/brug/ui/ChatActivity.kt [hunk 4]
```diff
@@ -55,7 +54,7 @@ class ChatActivity : AppCompatActivity() {
     private lateinit var convID: String
 
     private lateinit var buttonSendTextMessage: ImageButton
-    private lateinit var recordButton: RecordButton
+    private lateinit var recordButton: ImageButton
```
Comment (pmcintyr) [inline app/src/main/java/com/github/brugapp/brug/ui/ChatActivity.kt:57]: Could you explain to me why we changed the Button type here? Thanks!!

**Review by pmcintyr — APPROVED** [review]

File: app/src/main/java/com/github/brugapp/brug/ui/ChatActivity.kt (hunk 1 above)
Comment (hamzahass) [reply app/src/main/java/com/github/brugapp/brug/ui/ChatActivity.kt:253]: For the setListenForRecord function I just deleted for two reasons, the first being that I stopped using the RecordView layout for the audios (and setListenForRecord was I function of that library), but also we don't need to set for listen, as when we click on the button to record, the button disappears (visibility set to gone) and the buttons to stop and delete the audio show up.

File: app/src/test/java/com/github/brugapp/brug/MessageTest.kt (hunk 2 above)
Comment (hamzahass) [reply app/src/test/java/com/github/brugapp/brug/MessageTest.kt:250]: Actually we can, but it's just to have a better understanding of the code. In that case the url and the path are the same, but in some cases it isn't, so I figured it was better to keep them separate.

File: app/src/test/java/com/github/brugapp/brug/MessageTest.kt (hunk 3 above)
Comment (hamzahass) [reply app/src/test/java/com/github/brugapp/brug/MessageTest.kt:267]: The reason is the same as the other

**Review by ha2san — APPROVED** [review]
Really nice catch, this shows that just because everything works on the emulator does not mean that there are no problems. Well done

**Pull request merged**